File: client/lib/models/place.dart
import 'package:flutter_ngekos/models/city.dart';
import 'package:flutter_ngekos/models/country.dart';
import 'package:flutter_ngekos/models/feature.dart';
import 'package:flutter_ngekos/models/photo.dart';

class Place {
  int? id;
  String? name;
  String? description;
  int? price;
  int? rating;
  String? address;
  String? phone;
  String? imageUrl;
  String? mapUrl;
  City? city;
  Country? country;
  Feature? feature;
  List<Photo>? photo;
  bool isFavorite = false;

  Place(
      {this.id,
      this.name,
      this.description,
      this.price,
      this.rating,
      this.address,
      this.phone,
      this.imageUrl,
      this.mapUrl,
      this.city,
      this.country,
      this.feature,
      this.photo,
      this.isFavorite = false});

  Place.fromJson(Map<String, dynamic> json) {
    id = json['id'];
    name = json['name'];
    description = json['description'];
    price = json['price'];
    rating = json['rating'];
    address = json['address'];
    phone = json['phone'];
    mapUrl = json['mapUrl'];
    imageUrl = json['imageUrl'];
    city = json['city'] != null ? new City.fromJson(json['city']) : null;
    country =
        json['country'] != null ? new Country.fromJson(json['country']) : null;
    feature =
        json['feature'] != null ? new Feature.fromJson(json['feature']) : null;
    if (json['photo'] != null) {
      photo = <Photo>[];
      json['photo'].forEach((v) {
        photo!.add(new Photo.fromJson(v));
      });
    }
    isFavorite = json['isFavorite'];
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = new Map<String, dynamic>();

    data['id'] = this.id;
    data['name'] = this.name;
    data['description'] = this.description;
    data['price'] = this.price;
    data['rating'] = this.rating;
    data['address'] = this.address;
    data['phone'] = this.phone;
    data['mapUrl'] = this.mapUrl;
    if (this.city != null) {
      data['city'] = this.city!.toJson();
    }
    if (this.country != null) {
      data['country'] = this.country!.toJson();
    }
    if (this.feature != null) {
      data['feature'] = this.feature!.toJson();
    }
    if (this.photo != null) {
      data['photo'] = this.photo!.map((v) => v.toJson()).toList();
    }
    data['isFavorite'] = this.isFavorite;

    return data;
  }
}

File: server/models/Place.js
const Sequelize = require("sequelize");
const db = require("../config/database");
const City = require("./City");
const Country = require("./Country");
const Feature = require("./Feature");
const Photo = require("./Photo");

const Place = db.define(
    "places",
    {
        name: { type: Sequelize.STRING },
        description: { type: Sequelize.STRING },
        city_id: { type: Sequelize.INTEGER },
        country_id: { type: Sequelize.BIGINT },
        price: { type: Sequelize.INTEGER },
        rating: { type: Sequelize.INTEGER },
        imageUrl: { type: Sequelize.TEXT },
        address: { type: Sequelize.STRING },
        phone: { type: Sequelize.STRING },
        mapUrl: { type: Sequelize.STRING },
        isFavorite: { type: Sequelize.BOOLEAN },
    },
    {
        freezeTableName: true
    }
);

Place.belongsTo(City, { as: "city", foreignKey: 'city_id'});

Place.belongsTo(Country, { as: "country", foreignKey: 'country_id'});

Place.belongsTo(Feature, { as: "feature", foreignKey: 'featureId'});

Place.hasMany(Photo, {as: "photo"});

module.exports = Place;